assets/js/amber.js

```diff
@@ -226,3 +226,28 @@ export class Socket {
 module.exports = {
   Socket: Socket
 }
+
+
+/**
+ * Allows delete links to post for security and ease of use similar to Rails jquery_ujs
+ */
+document.addEventListener("DOMContentLoaded", function () {
+    document.querySelectorAll("a[data-method='delete']").forEach(function (element) {
```

Comment:
Why not make it more generic, not only for _delete_?

Reply:
Hi @Sija , what do you propose? :sweat_smile:

Reply:
Are there other cases where we would want a link to trigger a post?

Reply:
@Sija Ok, what about using this:

```js
document.querySelectorAll("a[data-method]").forEach(function (element) {
    var method = element.getAttribute("data-method");
    element.addEventListener("click", function (e) {
        e.preventDefault();
        var message = element.getAttribute("data-confirm") || "Are you sure?";
        if (confirm(message)) {
            var form = document.createElement("form");
            var input = document.createElement("input");
            form.setAttribute("action", element.getAttribute("href"));
            form.setAttribute("method", "POST");
            input.setAttribute("type", "hidden");
            input.setAttribute("value", method);
            form.appendChild(input);
            document.body.appendChild(form);
            form.submit();
        }
    });
});
```

^ I don't if amber support put or patch method overwrite, though :sweat_smile:

Reply:
Demo: https://jsfiddle.net/rgvsmhvy/1/ :tada:

Reply:
uhm, exactly that's what I had in mind, yet it could be expanded still, up to the point of porting [`rails-ujs`](https://github.com/rails/rails/tree/master/actionview/app/assets/javascripts/rails-ujs) - would you like that? I could prepare a PR with initial implementation if needed.

Reply:
> ^ I don't if amber support put or patch method overwrite, though 😅

Looks like it is: https://github.com/amberframework/amber/blob/f84fe8c4b43f9e0b307f0cd7ef1a5085e284e45a/spec/amber/router/request_spec.cr#L86-L91

Reply:
@Sija Oh, excellent, would be a nice PR :smile:

Reply:
@Sija I just opened an issue here: https://github.com/amberframework/amber/issues/738

Thank for the idea :+1: